src/common/enums/challenge.lifecycle.templates.ts

```diff
@@ -0,0 +1,3 @@
+export enum ChallengeLifecycleTemplates {
+  EXTENDED = 'EXTENDED',
```

Comment:
I am assuming extended will be common for all / mostly all templates - it might make sense to have lifecycle.templates.ts with LifecycleTemplates enum (essentially a 'common' for LifecycleTemplates) and have EXTENDED there.

Reply:
Actually I expect we could have many templates - not just a few. So we need later to have some kind of registration / factory setup for lifecycle definitions. To be brainstormed....there are two for now to illustrate the concept.